solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+
+@description('This is the Azure Virtual Network Manager which will be used to implement the connected group for spoke-to-spoke connectivity.')
+resource networkManager 'Microsoft.Network/networkManagers@2022-05-01' = {
+  name: 'avnm-${location}'
+  location: location
+  properties: {
+    networkManagerScopeAccesses: [
+      'Connectivity'
+      'SecurityAdmin'
+    ]
+    networkManagerScopes: {
+      subscriptions: [
+        '/subscriptions/${subscription().subscriptionId}'
+      ]
+      managementGroups: []
+    }
+  }
+  resource networkGroupProd 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-spokes-prod'
+    properties: {
+      description: 'Prod Spoke VNETs Network Group'
+    }
+    resource staticMembersSpokeOne 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: prodNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembersSpokeTwo 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: prodNetworkGroupMembers[1]
+      }
+    }
+  }
+  resource networkGroupNonProd 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-spokes-nonprod'
+    properties: {
+      description: 'Non-prod Spoke VNETs Network Group'
+    }
+    resource staticMembersSpokeOne 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembersSpokeTwo 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+  }
+  resource networkGroupAll 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-all'
+    properties: {
+      description: 'All VNETs Network Group (for Security Configurations)'
+    }
+    resource staticMembers1 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-0'
+      properties: {
+        resourceId: prodNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers2 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: prodNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers3 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers4 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-3'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers5 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-4'
+      properties: {
+        resourceId: hubVnetId
+      }
+    }
+  }
+}
+
+@description('This connectivity configuration defines the connectivity between the spokes. Only deployed if requested.')
+resource connectivityConfigurationNonProd 'Microsoft.Network/networkManagers/connectivityConfigurations@2022-05-01' = {
```

Comment:
Comment here implies it's conditional at this point.  But it doesn't appear to be.